### PR TITLE
Make ResultColumnMapper lambda-friendly

### PR DESCRIPTION
**Pull request by qualidafial**

### CODE REVIEW

**Comment by qualidafial** [comment]
@stevenschlansker could you code review this?

**Comment by stevenschlansker** [comment]
Instead of removing the `String` version of the methods, what would you say to having a default implementation that does the column lookup and delegates?

**Comment by stevenschlansker** [comment]
Another option, leave the interface as is, and have a subinterface that defaults the unused method.  Then we leave the current state as is and allow people to have a functional implementation of the subinterface?

**Comment by qualidafial** [comment]
That's a good idea. If we make the String variant a default method, it can still be a functional interface.

**Comment by qualidafial** [comment]
Hmm, while I'm at it, I think I'll use a similar trick to what you did in BuiltInArgumentFactory and get rid of the need for all the primitive column mapper classes.

**Comment by qualidafial** [comment]
@stevenschlansker Can you take another look now?

**Comment by stevenschlansker** [comment]
Whoa, net -500 lines.  I'm excited :)

File: core/src/main/java/org/jdbi/v3/PrimitivesMapperFactory.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import org.jdbi.v3.util.EnumMapper;
+
+/**
+ * Result column mapper factory which knows how to map standard JDBC-recognized types.
+ */
+public class PrimitivesMapperFactory implements ResultColumnMapperFactory {
+    private static final Map<Class<?>, ResultColumnMapper<?>> mappers = new HashMap<>();
+
+    static {
+        mappers.put(boolean.class, primitiveMapper(ResultSet::getBoolean));
```
Comment (stevenschlansker) [inline core/src/main/java/org/jdbi/v3/PrimitivesMapperFactory.java]: This makes me very happy :smile:

**Pull request merged**